DMX.Gatekeeper.Api.Tests.Acceptance/APIs/Labs/LabApiTests.cs

```diff
@@ -38,104 +39,7 @@ public LabApiTests(DmxGatekeeperApiBroker dmxGatekeeperApiBroker)
-                    Name = GetRandomString(),
-                    Description = GetRandomString(),
-                    Status = LabStatus.Reserved,
-                    Devices = randomDevices
-                },
-
-                new Lab
-                {
-                    Id = Guid.NewGuid(),
-                    ExternalId = GetRandomNumber().ToString(),
-                    Name = GetRandomString(),
-                    Description = GetRandomString(),
-                    Status = LabStatus.Offline,
-                    Devices = randomDevices
-                }
-            };
-
-            return Enumerable.Range(start: 0, count: randomCount)
-                .Select(iterator => allCases)
-                    .SelectMany(@case => @case)
-                        .ToList();
-        }
-
-        private static List<LabDevice> GetRandomLabDevices()
-        {
-            string randomPhoneName = GetRandomString();
-            string randomHMDName = GetRandomString();
-            bool randomHostConnectionStatus = GetRandomBoolean();
-            bool randomPhoneConnectionStatus = GetRandomBoolean();
-            bool randomHMDConnectionStatus = GetRandomBoolean();
-            int randomPhonePowerLevel = GetRandomPowerLevel();
-            int randomHMDPowerLevel = GetRandomPowerLevel();
-
-            List<LabDevice> labDevices = new List<LabDevice>
-            {
-                new LabDevice
-                {
-                    Id = Guid.NewGuid(),
-                    Name = null,
-                    Type = LabDeviceType.PC,
-                    Category = LabDeviceCategory.Host,
-                    PowerLevel = null,
-
-                    Status = randomHostConnectionStatus
-                        ? LabDeviceStatus.Online
-                        : LabDeviceStatus.Offline,
-                },
-
-                new LabDevice
-                {
-                    Id = Guid.NewGuid(),
-                    Name = randomPhoneName,
-                    Type = LabDeviceType.Phone,
-                    Category = LabDeviceCategory.Attachment,
-                    PowerLevel = randomPhonePowerLevel,
-
-                    Status = randomPhoneConnectionStatus
-                        ? LabDeviceStatus.Online
-                        : LabDeviceStatus.Offline,
-                },
-
-                new LabDevice
-                {
-                    Id = Guid.NewGuid(),
-                    Name = randomHMDName,
-                    Type = LabDeviceType.HeadMountedDisplay,
-                    Category = LabDeviceCategory.Attachment,
-                    PowerLevel = randomHMDPowerLevel,
-
-                    Status = randomHMDConnectionStatus
-                        ? LabDeviceStatus.Online
-                        : LabDeviceStatus.Offline,
-                },
-            };
-
-            return labDevices;
-        }
+        private static Filler<Lab> CreateLabFiller() =>
+            new Filler<Lab>();
```

Comment:
@terrypalo not in the acceptance tests case. Acceptance tests are not about business logic or scenarios. they focus mostly on making sure your controller endpoint is talking to the right service and your service is connected to the right external resource through their brokers.

Unit tests are the ones who care more about every single scenario and such.

Reply:
Gotcha. Curious as to why we don't just fill an external device and lab device in Core? This would also simplify the acceptance tests in there as well

Reply:
Because of the mapping. If you auto-fill these two your unit tests would be repeating the same logic as your service.  This brings us back to Euclid's Law of Axioms - _things that are equal to the same thing are equal to each other_

Reply:
This is a good reference to this testing methedology:
https://www.youtube.com/watch?v=sOqcMRn-HSc

Reply:
I think I understand.. The "scenarios" I mentioned [here](https://github.com/microsoft/mixedreality.dmx.core/blob/176f8f13509a16deb540d63d112d196c25c6672a/DMX.Core.Api.Tests.Acceptance/APIs/Labs/LabApiTests.cs#L109) arent actually scenarios-- They're just a mapping for ExternalLabs and Labs, so there is an accurate continuity between the two